### ISSUE TITLE
application.js contains huge list of translations

### ISSUE THREAD

**Issue by dawidi** [openstreetmap/openstreetmap-website#1021]
An anonymous visitor to #openstreetmap on IRC pointed out that the slippymap contains a huge javascript file: assets/application-....js currently takes over 600 KB to download (gzipped), and expands to nearly 3 megabytes of Javascript.
This may not be "that much", compared to other map services, and you won't notice it unless you have a slow connection (or worse, a slow computer), but it bugs me especially because I otherwise like the OSM slippymap for being relatively lightweight.

90% of that javascript file is an associative array of translations (of labels, UI elements, messages, and tag/value descriptions) in 120+ languages, about 20 KB for each language.
Most clients will only ever use values from one or maybe two languages, the rest just puts unnecessary load on the network and browser.

Those translations could be put into separate files for each language (presumably that code is auto-generated anyway?), and the main script would then only have to download the one language that it actually plans to use on this client.

**Comment by tomhughes** [comment]
The i18n-js module that we use for javascript internationalisation doesn't support this at the moment unfortunately, at least not when used via the asset pipeline.

**Comment by tomhughes** [comment]
https://github.com/fnando/i18n-js/issues/343 is an upstream bug covering this issue.

**Comment by AntiFish03** [comment]
If you want take a look at fnando/i18n-js#354.  I added functionality that would allow multiple separate translation files to be loaded.

**Comment by tomhughes** [comment]
@AntiFish03 Thanks for the pointer, but I don't think that's really doing what I want. What I'm looking to do is to split by language rather than route - so have precompile generate i18n/en.js and so on and then have my application load the right js.

**Comment by AntiFish03** [comment]
which can be done with and export

```yaml
translations:
- file: "public/javascripts/i18n/%{locale}.js"
  only: '*'

```

which will break it out by each locale it finds.

**Comment by tomhughes** [comment]
Right, but you have to manually export them because the asset pipeline won't pick those up. Well not unless you copy the `i18n/translations.js` file in the i18n-js assets to each of the other names?

I need to investigate and see if there is a way for i18n-js to tell sprockets it is going to handle names without actually having a physical file on disk for them...

**Comment by mmd-osm** [comment]
This issue can be closed, fix was implemented in #1949
